server/render/index.js

```diff
@@ -226,3 +199,52 @@ export function serverRenderError( err, req, res, next ) {
 
 	next();
 }
+
+/**
+ * Middlewares may decide to override (or add, or remove) the validators defined as defaults.
+ *
+ * Adds SSRValidators to the context, an object in which property names are idnetifiers and the values are callables
+ * that shouls return booleans. If one of the validators fails, then SSR is considered disabled. All methdos would
+ * receive the context on call.
+ *
+ * @param {object}   context  The entire request context
+ * @param {function} next     As all middlewares, will just add next
+ */
+export function initShouldSSR( context, next ) {
+	context.SSRValidators = {
+		config: () => config.isEnabled( 'server-side-rendering' ),
+		section: () => isSectionIsomorphic( context.store.getState() ),
+		user: () => ! context.user, // logged out
+		locale: () => isDefaultLocale( context.lang ),
+		layout: () => context.layout,
+		query: () => Object.keys( context.query ).length === 0, // no query args
+	};
```

Comment:
> Some of these conditions seem relevant for middlewares, some are app level. The app level ones might as well live in a centralized place and not let middlewares touch them.

This :+1:

Reply:
> … I wonder about this one. Any reason not to SSR other locales?
Can't do that right now, because `i18n-calypso` stores the current locale at module level. In a server setting, this would lead to cross-request pollution. Arguably something we should fix at some point, tho.